test/unit/test_utils.py

```diff
@@ -60,3 +63,32 @@ def test_mysql_dump_processor_remove_log_bin(line):
 def test_mysql_dump_processor_remove_definers(line_in, line_out):
     helper = MySQLDumpProcessor()
     assert helper.process_line(line_in) == line_out
+
+
+@mark.parametrize(
+    "uri, exception_class",
+    [
+        ("mysql://<user>:<pwd>@<ip>:1234/", None),
+        ("mysql://<user>:<pwd>@<ip>:1234/?", None),
+        ("mysql://<user>:<pwd>@<ip>:1234/?ssl-mode=DISABLE", None),
+        ("mysql://<user>:<pwd>@<ip>:1234/?ssl-mode=REQUIRE", None),
+        # options with no values get dropped
+        ("mysql://<user>:<pwd>@<ip>:1234/?ssl-mode=", None),
+        # extra parameter
+        ("mysql://<user>:<pwd>@<ip>:1234/?ssl-mode=REQUIRE&bar=baz", WrongMigrationConfigurationException),
+        # unsupported value for ssl-mode
+        ("mysql://<user>:<pwd>@<ip>:1234/?ssl-mode=something", WrongMigrationConfigurationException),
+        # unexpected parameter
+        ("mysql://<user>:<pwd>@<ip>:1234/?foo=bar", WrongMigrationConfigurationException),
+        # passing the ssl-mode twice
+        ("mysql://<user>:<pwd>@<ip>:1234/?ssl-mode=REQUIRE&ssl-mode=DISABLE", WrongMigrationConfigurationException),
```

Comment:
Even though RFC does not specify what happens in this case, I think it's good to raise an error here to be explicit